freeipa-api/src/main/java/com/sequenceiq/freeipa/api/v1/KerberosMgmt/KerberosMgmtV1Endpoint.java

```diff
@@ -0,0 +1,50 @@
+package com.sequenceiq.freeipa.api.v1.KerberosMgmt;
+
+import javax.validation.Valid;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.doc.KeytabModelNotes;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.doc.KeytabOperationsDescription;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.model.ServiceKeytabRequest;
+import com.sequenceiq.freeipa.api.v1.KerberosMgmt.model.ServiceKeytabResponse;
+import com.sequenceiq.service.api.doc.ContentType;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+
+@Path("/v1/kerberosmgmt")
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(value = "/v1/kerberosmgmt", protocols = "http,https")
+public interface KerberosMgmtV1Endpoint {
+
+    @POST
+    @Path("createprincipal")
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(value = KeytabOperationsDescription.DESCRIBE_CREATE_HOST_PRINCIPAL, produces = ContentType.JSON,
+            notes = KeytabModelNotes.CREATE_HOST_AND_PRINCIPAL_NOTES,
+            nickname = "getServiceKeytabV1")
+    void createHostAndPrincipal(@Valid ServiceKeytabRequest request);
+
+    @GET
+    @Path("servicekeytab")
+    @ApiOperation(value = KeytabOperationsDescription.DESCRIBE_SERVICE_KEYTAB, produces = ContentType.JSON, notes = KeytabModelNotes.GET_KEYTAB_NOTES,
+            nickname = "getServiceKeytabV1")
+    ServiceKeytabResponse getServiceKeytab(@Valid ServiceKeytabRequest request);
+
+    @DELETE
+    @Path("serviceprincipal")
+    @ApiOperation(value = KeytabOperationsDescription.DESCRIBE_DELETE_SERVICE_PRINCIPAL, notes = KeytabModelNotes.DELETE_SERVICE_PRINCIPAL_NOTES,
+            nickname = "deleteServicePrinciapalV1")
+    void deleteServicePrincipal(@Valid ServiceKeytabRequest request);
+
+    @DELETE
+    @Path("host")
+    @ApiOperation(value = KeytabOperationsDescription.DESCRIBE_DELETE_HOST, notes = KeytabModelNotes.DELETE_HOST_NOTES, nickname = "deleteHostV1")
+    void deleteHost(@Valid ServiceKeytabRequest request);
```

Comment:
Should the param be a host request (there are fewer details needed than the service keytab request)?

Reply:
done.